sql/rowexec/agg.go

```diff
@@ -248,7 +249,10 @@ func groupingKey(
 
 		t, isStringType := expr.Type().(sql.StringType)
 		if isStringType && v != nil {
-			err = t.Collation().WriteWeightString(hash, v.(string))
+			v, err = types.ConvertToString(v, t)
+			if err == nil {
```

Comment:
should anything happen if err != nil? are we dropping an error or is it impossible for this to map to an aggregation group

Reply:
if `err != nil`, it should fall through to the if [below the else](https://github.com/dolthub/go-mysql-server/pull/1844/files/cc2ddbc8bde618b220c4bebfd45149741dd40e1f#diff-dad61c9d4929b839c2f57e8e0739bc4709607c0f62e4c8737c045592d6a1f046R259), and `return 0, err`